### ISSUE TITLE
Groups with same name not sharing errors

### ISSUE THREAD

**Issue by jjjrmy** [formkit/formkit#868]
### Reproduction

https://formkit.link/cc52fafd117863912a545caff74f1cf6

### Describe the bug

the forms data is sending as an object with all fields under the group (expected)
but the 2nd group of the same name is not displaying the errors

### Environment

Latest

**Comment by GustavoFenilli** [comment]
@justin-schroeder the behaviour is inconsistent, not sure if its a bug on what side of the problem.

1. It should replace/ignore name conflicts
2. Or it should combine errors too

Also as a work around the problem you can manually pass combined names to the same node so it works: https://formkit.link/707cf8f212efc1e72b3e9dae0f42867c

**Comment by justin-schroeder** [comment]
Yeah — names are meant to be unique. When it searches the tree for a node by a given name it does a BFS — in otherwords it stops the search when a result is found. Not sure we are interested in reversing course on that fundamental distinction.

Errors are not the only thing that would be affected by identical names. You’ll have issues with hydration, state, resets and more.

Are there good use cases for identical input names we have not considered?

**Comment by jjjrmy** [comment]
> Are there good use cases for identical input names we have not considered?

Specifically just GROUPS should be allowed to be multiple. I can see that it may cause some unexpected problems.

But the reason I came into this issue is because I have Inputs that have "grouped names" (as in their errors are referenced by `group.foo`, and `group.bar`) but are not in the same location to be put into a FormKit Group. So the only way to show their errors would be to place them in the groups like the original example (or use my custom plugin to allow any input to be put into a group it is not a child of).

**Comment by pedromelo222** [comment]
I have the same issue. It would be great to have this implemented in FormKit.

I work on a SAS project in the insurance sector, and we have over 20 massive and complex forms. In all the forms, we have inputs that are scattered on the screen but belong to the same group. Most of them are positioned in a way that makes it easy for the user to understand what they relate to.

I tested FormKit, and it was by far the best solution I found for transforming forms into a schema list. Grouping separate fields was one of the first things I checked to see if it met my needs (and it worked... until I reached the validation part).

Right now, I'm trying to find a solution to group separate fields under the same object in a way that works with Zod validation. If anyone has any ideas or suggestions, I would appreciate it!

@jjjrmy If possible, could you share the plugin you developed to group the inputs into the same object?

**Comment by gerbenvandijk** [comment]
Enterprise customer here.

I am too running into this issue, was quite the journey in regards of time to end up on this issue. Same use case.

Another example of this issue: https://formkit.link/1dfa6a5e7ba350b3591305b01b62f693

For me, @fenilli his plugin solution works fine. However it would be great if this could become part of FormKit core.
